src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig

```diff
@@ -161,18 +162,39 @@
                     });
                 });
 
-                /* Prevents from disallowing clicks on "copy to clipboard" elements inside toggles */
-                const copyToClipboardElements = toggle.querySelectorAll('span[data-clipboard-text]');
-                copyToClipboardElements.forEach((copyToClipboardElement) => {
-                    copyToClipboardElement.addEventListener('click', (e) => {
-                        e.stopPropagation();
-                    });
-                });
-
                 toggle.setAttribute('data-processed', 'true');
             });
         }
 
+        #createCopyToClipboard() {
+            if (!navigator.clipboard) {
+                return;
+            }
+
+            const copyToClipboardElements = document.querySelectorAll('[data-clipboard-text]');
+
+            copyToClipboardElements.forEach((copyToClipboardElement) => {
+                copyToClipboardElement.classList.remove('hidden');
+
+                copyToClipboardElement.addEventListener('click', (e) => {
+                    /* Prevents from disallowing clicks on "copy to clipboard" elements inside toggles */
+                    e.stopPropagation();
+
+                    navigator.clipboard.writeText(copyToClipboardElement.getAttribute('data-clipboard-text'));
+
+                    let oldContent = copyToClipboardElement.textContent;
+
+                    copyToClipboardElement.textContent = `✅ Copied!`;
+                    copyToClipboardElement.classList.add('status-success');
```

Comment:
Excellent improvement over the previous logic. One tiny issue: If you're using the dark theme, the text color will be black, which makes it hard/impossible to read. The default state itself is fine, just the `status-success` state needs a `color`-update.

Default state:

![CleanShot 2023-12-08 at 22 20 01@2x](https://github.com/symfony/symfony/assets/439899/6560329f-952a-4e9f-a0dd-8dd237196fd1)

With focus state:

![CleanShot 2023-12-08 at 22 19 37@2x](https://github.com/symfony/symfony/assets/439899/0364cfeb-3293-40ad-9765-d60969557afd)

Without focus state:

![CleanShot 2023-12-08 at 22 19 39@2x](https://github.com/symfony/symfony/assets/439899/ba8e96f9-ac6e-4d36-9c42-a0c2c1923653)

Reply:
Thanks for the notice, unfortunately, I'm not sure about the expected design here. Maybe @javiereguiluz can give us a clue.

Reply:
I'd say we don't need to change the color when copying things.

Just changing the button text to "Copied" is enough. That's e.g. what GitHub does, so everybody is used to that behavior.

Thanks.